bond/commands/backup.py

```diff
@@ -0,0 +1,189 @@
+        timestamp = str(int(time.time()))
+        body = {
+            "backup": 1,
+            "http_port": "4444",
+            "path": "",
+            "timestamp": timestamp,
+        }
+        rsp = bond.proto.put(bondid, topic="sys/backup", body=body)
+        print(rsp)
+        if rsp["s"] != 200:
+            raise Exception(
+                "Error HTTP %d starting backup: %s", rsp["s"], rsp["b"]["_error_msg"]
+            )
+        time.time()
+        while True:
+            time.sleep(1)
+            rsp = bond.proto.get(bondid, topic="sys/backup")
+            if rsp["b"]["backup"] == -1:
+                raise Exception("Backup error: %s" % rsp["b"]["error_msg"])
+            if rsp["b"]["backup"] == 2:
+                print("Bond reports backup success.")
+                break
+        path = ""
+        while timestamp not in path:
+            path = wait_upload(5)
+        print("Backup saved in ~/.bond directory")
+
+
+def get_file_list():
+    fns = list(map(str, Path(DB_DIRNAME).glob("**/*.bondbackup")))
+    rv = []
+    for fn in fns:
+        tok = fn.split("_")
+        rv.append(
+            {
+                "file": fn,
+                "bondid": tok[0],
+                "version": tok[1],
+                "timestamp": int(tok[2]),
+                "datestr": datetime.datetime.utcfromtimestamp(int(tok[2]))
+                .astimezone()
+                .strftime("%Y-%m-%d %H:%M:%S"),
+            }
+        )
+    rv.sort(key=lambda x: x["timestamp"], reverse=True)
+    return rv
+
+
+class RestoreCommand(BaseCommand):
+    subcmd = "restore"
+    help = """Restore a Bond."""
+    arguments = {
+        "--list": {
+            "help": "list backups available for restore to selected bond",
+            "action": "store_true",
+        },
+        "--file": {
+            "help": "filename of backup to restore (must be in ~/.bond/)",
+        },
+        "--latest": {
+            "help": "restore the most recent backup for selected bond",
+            "action": "store_true",
+        },
+        "--no-reboot": {
+            "help": "Only for test purposes. May cause unexpected behavior.",
+            "action": "store_true",
+        },
+    }
+
+    def run(self, args):  # noqa: C901
+        file_list = get_file_list()
+        if args.list or not (args.file or args.latest) or len(file_list) == 0:
+            if len(file_list) == 0:
+                print("No backups found")
+            else:
+                print("Found %d backups: " % len(file_list))
+                for f in get_file_list():
+                    print("  " + f["datestr"] + "   " + f["file"])
+            return
+
+        if args.latest:
+            if len(file_list) == 0:
+                print("No backups found")
+                return
+            args.file = file_list[-1]["file"]
+
+        start_daemon(4444)
+        bondid = BondDatabase.get_assert_selected_bondid()
+        # timestamp = str(int(time.time()))
```

Comment:
are we going to use this timestamp? I commented it out since the linter was complaining about an unused variable